messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -397,6 +415,22 @@ public Builder messageMonitor(MessageMonitor<? super QueryMessage<?, ?>> message
             return this;
         }
 
+        /**
+         * Sets the duplicate query handler resolver. This determines which registrations are added to the query bus
+         * when multiple handlers are detected for the same query.
+         * <p>
+         * {@link DuplicateQueryHandlerResolution} contains good examples on this and will most of the time suffice. The
+         * bus defaults to {@link DuplicateQueryHandlerResolution#logAndAccept()}
+         *
+         * @param duplicateQueryHandlerResolver The {@link} DuplicateQueryHandlerResolver to use when multiple
+         *                                      registrations are detected
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder duplicateQueryHandlerResolver(DuplicateQueryHandlerResolver duplicateQueryHandlerResolver) {
```

Comment:
I'd expect a validation that the provided resolver isn't null. `BuilderUtils#assertNonNull` will suffice in this case.

Or, when we're using the provided `duplicateQueryHandlerResolver` we should check whether it's null.